tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
-        result = self.space.metric.exp(vec, initial_point)
-        self.assertAllClose(end_point, result)
-
-    def test_log_extreme_case(self):
-        initial_point = self.space.random_uniform(2)
-        vec = 1e-4 * gs.random.rand(*initial_point.shape)
-        vec = self.space.to_tangent(vec, initial_point)
-        point = self.space.metric.exp(vec, base_point=initial_point)
-        result = self.space.metric.log(point, initial_point)
-        self.assertAllClose(vec, result)
-
-    def test_exp_and_log_and_projection_to_tangent_space_general_case(self):
-        """Test Log and Exp.
-
-        Test that the Riemannian exponential
-        and the Riemannian logarithm are inverse.
-
-        Expect their composition to give the identity function.
-
-        NB: points on the n-dimensional sphere are
-        (n+1)-D vectors of norm 1.
-        """
-        # Riemannian Exp then Riemannian Log
-        # General case
-        # NB: Riemannian log gives a regularized tangent vector,
-        # so we take the norm modulo 2 * pi.
-        base_point = gs.array([0.0, -3.0, 0.0, 3.0, 4.0])
-        base_point = base_point / gs.linalg.norm(base_point)
-
-        vector = gs.array([3.0, 2.0, 0.0, 0.0, -1.0])
-        vector = self.space.to_tangent(vector=vector, base_point=base_point)
-
-        exp = self.metric.exp(tangent_vec=vector, base_point=base_point)
-        result = self.metric.log(point=exp, base_point=base_point)
-
-        expected = vector
-        norm_expected = gs.linalg.norm(expected)
-        regularized_norm_expected = gs.mod(norm_expected, 2 * gs.pi)
-        expected = expected / norm_expected * regularized_norm_expected
-
-        # The Log can be the opposite vector on the tangent space,
-        # whose Exp gives the base_point
-        are_close = gs.allclose(result, expected)
-        norm_2pi = gs.isclose(gs.linalg.norm(result - expected), 2 * gs.pi)
-        self.assertTrue(are_close or norm_2pi)
-
-    def test_exp_and_log_and_projection_to_tangent_space_edge_case(self):
-        """Test Log and Exp.
-
-        Test that the Riemannian exponential
-        and the Riemannian logarithm are inverse.
-
-        Expect their composition to give the identity function.
-
-        NB: points on the n-dimensional sphere are
-        (n+1)-D vectors of norm 1.
-        """
-        # Riemannian Exp then Riemannian Log
-        # Edge case: tangent vector has norm < epsilon
-        base_point = gs.array([10.0, -2.0, -0.5, 34.0, 3.0])
-        base_point = base_point / gs.linalg.norm(base_point)
-        vector = 1e-4 * gs.array([0.06, -51.0, 6.0, 5.0, 3.0])
-        vector = self.space.to_tangent(vector=vector, base_point=base_point)
-
-        exp = self.metric.exp(tangent_vec=vector, base_point=base_point)
-        result = self.metric.log(point=exp, base_point=base_point)
-        self.assertAllClose(result, vector)
-
-    def test_squared_norm_and_squared_dist(self):
```

Comment:
issing? this could become a test in the RiemannianMetricParametrizer.

Reply:
already part of improvs